src/pages/transactions/edit/Edit.tsx

```diff
@@ -0,0 +1,207 @@
+import {
+  defaultTransactionProperties,
+  transactionTypes,
+} from 'common/constants/transactions';
+import { useCurrencies } from 'common/hooks/useCurrencies';
+import { useTitle } from 'common/hooks/useTitle';
+import { endpoint } from 'common/helpers';
+import { TransactionInput } from 'common/interfaces/transactions';
+import { Container } from 'components/Container';
+import { DebouncedCombobox } from 'components/forms/DebouncedCombobox';
+import { Default } from 'components/layouts/Default';
+import { useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useTransactionPages } from '../common/hooks/useTransactionPages';
+import { useTransactionValidation } from '../common/hooks/useTransactionValidation';
+import { TransactionValidation } from '../common/validation/ValidationInterface';
+import { request } from 'common/helpers/request';
+import toast from 'react-hot-toast';
+import { useNavigate, useParams } from 'react-router-dom';
+import { route } from 'common/helpers/route';
+
+const Edit = () => {
+  const { t } = useTranslation();
+  const navigate = useNavigate();
+  const { id } = useParams<string>();
+  const currencies = useCurrencies();
+  const pages = useTransactionPages();
+  const { documentTitle } = useTitle('edit_transaction');
+  const [isSaving, setIsSaving] = useState<boolean>(false);
+  const [isLoading, setIsLoading] = useState<boolean>(true);
+  const [errors, setErrors] = useState<TransactionValidation>();
+  const [transaction, setTransaction] = useState<TransactionInput>(
+    defaultTransactionProperties
+  );
+  const { checkValidation, setValidation } = useTransactionValidation();
+
+  const getTransactionInputObject = (
+    responseDetails: any
+  ): TransactionInput => {
+    const { date, amount, currency_id, bank_integration_id, description } =
+      responseDetails || {};
+    return {
+      base_type:
+        responseDetails?.base_type === 'CREDIT' ? 'deposit' : 'withdrawal',
+      date,
+      amount,
+      currency_id,
+      bank_integration_id,
+      description,
+    };
+  };
+
+  const fetchTransactionDetails = async () => {
+    const response = await request(
+      'GET',
+      endpoint('/api/v1/bank_transactions/:id', { id })
+    );
+    setIsLoading(false);
+    setTransaction(getTransactionInputObject(response?.data?.data));
+  };
+
+  const handleChange = (
+    property: keyof TransactionInput,
+    value: unknown
+  ): void => {
+    setTransaction(
+      (transaction) => transaction && { ...transaction, [property]: value }
+    );
+  };
+
+  const onSave = async (event: React.FormEvent<HTMLFormElement>) => {
+    event?.preventDefault();
+    setErrors(undefined);
+    const isFormValid = checkValidation(transaction);
+
+    if (!isFormValid && !errors) {
+      setValidation(transaction, setErrors, errors);
+    }
+
+    if (isFormValid) {
+      setIsSaving(true);
+      const toastId = toast.loading(t('processing'));
```

Comment:
Please make use of `toast` within the app itself.

Reply:
> Please make use of `toast` within the app itself.

done